extensions/image-preview/src/preview.ts

```diff
@@ -120,6 +120,7 @@ export class Preview extends Disposable {
 			}
 			this._previewState = PreviewState.Visible;
 		}
+		this.webviewEditor.webview.postMessage({ type: 'setActive', value: this.webviewEditor.active });
```

Comment:
Instead of requiring message passing, also see if we can use `document.hasFocus()`  inside the webview. I think this may track what we want.

Reply:
Hi,

I have experimented a bit with ```document.hasFocus()``` and the ```'focus'``` event. ```document.hasFocus()``` is always set to true before any ```MouseEvent``` is fired, this makes it so that we can not use it to check if click was done without focus, same goes for the ```'focus'``` event. This can be solved by setting a state variable in ```focus``` event handler that blockes the first click after focus. However, this introduces a bug where if focus is gained without the use of a click, the user will be forced to click twice in order to zoom. I have not found any way to gain information about what caused the ```FocusEvent```, which would be needed to solve this bug.

When using message passing instead, the ```'message'``` event handler will be triggered after the ```'mousedown'``` event handler, but before the ```'click'``` event handler, if focus was gained through a click. This is useful, because it allows us to identify what caused the gain of focus, and to handle each case accordingly. In my solution the ```'mousedown'``` event handler changes the state, to allow a zoom to be executed only in the case where a click is initiated after focus was gained. For this this reason I will mark all as resolved.